system/mavros2/mavros_setup.sh

```diff
@@ -0,0 +1,126 @@
+else
+    export VEHICLE_NAMESPACE="vehicle_$MAVROS_TGT_SYSTEM"
+    echo "VEHICLE_NAMESPACE not set, default to auto generated default based on mavros target ($MAVROS_TGT_SYSTEM) of $VEHICLE_NAMESPACE"
+fi
+
+
+if [ -v $MAVROS_FCU_URL ]; then
+    # If not set, then generate automatically
+    if ! [ $HAS_VEHICLE_CONFIG = true ]; then
+        # No vehicle.config, generate based on subparameters and the target system id dynamically
+        INSTANCE=$((MAVROS_TGT_SYSTEM - 1)) # Default is based on Mavros TGT System
+        if [ $TGT_SYSTEM_SET = false ]; then # If Mavros TGT System was set automatically, use get_instance.
+            set +e
+            INSTANCE=$(get_instance)
+            set -e
+        fi
+        MAVROS_FCU_URL="$MAVROS_FCU_CONN://$MAVROS_FCU_IP:$((MAVROS_FCU_UDP_BASE + INSTANCE))@/"
+        export MAVROS_FCU_URL=$MAVROS_FCU_URL
+        echo "MAVROS_FCU_URL automatically set to: $MAVROS_FCU_URL"
+    else
+        # Use value from vehicle.config
+        export MAVROS_FCU_URL=$VEHICLE_FCU_URL
+        echo "MAVROS_FCU_URL set to: $MAVROS_FCU_URL from vehicle.config"
+    fi
+else
+    echo "MAVROS_FCU_URL was set to $MAVROS_FCU_URL"
+fi
+
+# Modify mavros PX4 config for frame ids
+MAVROS_MOD_CONFIG_PATH=${MAVROS_MOD_CONFIG_PATH:-"/mavros_config_mod.yaml"}
+MAVROS_CONFIG_PATH=${MAVROS_CONFIG_PATH:-"/mavros_config.yaml"}
+if [ ! -f "$MAVROS_MOD_CONFIG_PATH" ]; then
+    echo "Modified Mavros Config for $VEHICLE_NAMESPACE does not exist at $MAVROS_MOD_CONFIG_PATH. Generating specialised configuration for launch."
+    # This line:
+    # -E allow groupings defined by parenthesis to be used in replace by \1 \2
+    # /\"map\"/! selects lines which do not contain "map" or "earth" (Global frame)
+    # The rest then adds $VEHICLE_NAMESPACE in as a prefix to whatever the frame_id was
+    # This captures both frame_id: and child_frame_id
+    # This then saves the output into $PX4_MOD_CONFIG_PATH
+    sed -E "/\"map\"|\"earth\"/! s/(frame_id: )\"(.+)\"/\1\"$VEHICLE_NAMESPACE\/\2\"/g" $MAVROS_CONFIG_PATH > $MAVROS_MOD_CONFIG_PATH
+    sed -i "s#mavros#/$VEHICLE_NAMESPACE/mavros#g" $MAVROS_MOD_CONFIG_PATH
+else
+    echo "Modified Mavros Config for $VEHICLE_NAMESPACE already exists at $MAVROS_MOD_CONFIG_PATH. Using for launch"
+fi
+
+# Modify mavros PX4 config for frame ids
+MAVROS_MOD_PLUGINLISTS_PATH=${MAVROS_MOD_PLUGINLISTS_PATH:-"/mavros_pluginlists_mod.yaml"}
+MAVROS_PLUGINLISTS_PATH=${MAVROS_PLUGINLISTS_PATH:-"/mavros_pluginlists_px4.yaml"}
+if [ ! -f "$MAVROS_MOD_PLUGINLISTS_PATH" ]; then
+    echo "Modified Mavros Pluginlist for $VEHICLE_NAMESPACE does not exist at $MAVROS_PLUGINLISTS_PATH. Generating specialised configuration for launch."
+    sed -E "s#mavros#/$VEHICLE_NAMESPACE/mavros/mavros#g" $MAVROS_PLUGINLISTS_PATH > $MAVROS_MOD_PLUGINLISTS_PATH
+else
+    echo "Modified Mavros Pluginlist for $VEHICLE_NAMESPACE already exists at $MAVROS_MOD_PLUGINLISTS_PATH. Using for launch"
+fi
```

Comment:
[tangent] This is all PX4 specific, there probably needs to be ArduPilot versions of this somewhere too. At some point this probably becomes too complicated to trust to sketchy bash scripts. Might want to transition to Python.

Reply:
You are not wrong. But would probably take too much time. On Ardupilot, I have not actually tested it with Ardupilot and don't know how it would react! Might even need a similar bash code block for the ardupilot side of things too.

Reply:
Yeah this was to be broken out into a separate issue

Reply:
See #179